index.js

```diff
@@ -1,18 +1,47 @@
 const isObject = require('isobject');
 
+function parse(query) {
+  if (isObject(query)) {
+    return query;
+  }
+  try {
+    return JSON.parse(
+      query
+        .replace(/(?:\r\n|\r|\n)|,|:/g, ' ')
+        .replace(/([{}])/g, ' $1 ')
+        .replace(/ +/g, ' ')
+        .trim()
+        .replace(/([^ {}]+) /g, '"$1" ')
+        .replace(/" "/g, '":true, "')
+        .replace(/} "/g, '}, "')
+        .replace(/" {/g, '": {')
+        .replace(/" }/g, '":true }')
+    );
+  } catch (error) {
+    return null;
```

Comment:
We should log then re-throw the error - these errors will typically occur at parse time and should be fixed or it will lead to other issues.

```js
catch (error) {
  console.error(`Invalid objectql query: ${query}`);
  throw error;
}
```

Reply:
I set it to return null to avoid a definite API breaking change. With the current v1 API any `!isObject(query)` simply returns the source unaltered: https://github.com/bmullan91/objectql/blob/master/index.js#L4-L5

Are you certain you want to throw here? 

If so, how do you want to update README doc for:
`// invalid queries`